twill-core/src/main/java/org/apache/twill/kafka/client/KafkaConsumer.java

```diff
@@ -33,14 +33,16 @@
 
     /**
      * Invoked when new messages is available.
+     * @param startOffset Offset of the first {@link FetchedMessage} in the iterator of new messages.
```

Comment:
I understand the behavior, but is it a good API? As you see, it is quite difficult to explain, hence documenting it correctly. I feel it's much easier to explain and to use if we say the `startOffset` is the offset of the first message in the given iterator.